app/src/marketplace/containers/ProductPage/Hero.jsx

```diff
@@ -118,14 +118,17 @@ const Hero = () => {
     ])
 
     useEffect(() => {
-        const loadWhitelistStatus = async () => {
-            if (isWhitelistEnabled && productId && account) {
-                const whitelisted = await isAddressWhitelisted(productId, account)
-                setIsWhitelisted(whitelisted)
+        const loadWhitelistStatus = async (targetAccount) => {
+            let whitelisted = !isWhitelistEnabled
+            if (isWhitelistEnabled) {
+                whitelisted = await isAddressWhitelisted(productId, targetAccount)
             }
+            setIsWhitelisted(whitelisted)
```

Comment:
This reads a bit strange when formatted like this. After a minute I realized `whitelisted` starts off being true if whitelists are disabled. Otherwise it's false and we gotta verify product's "whitelistedness". In other words:

```js
const whitelisted = !isWhitelistEnabled || await isAddressWhitelisted(productId, targetAccount)
```

Hope I didn't miss anything important.

Reply:
Thanks, that's indeed the intention and more readable 👍 short-circuiting should take care that `isAddressWhitelisted` is called only when necessary